jsapp/js/components/processing/sidebar/sidebarDisplaySettings.tsx

```diff
@@ -158,6 +266,7 @@ export default function SidebarDisplaySettings() {
             color='light-blue'
             size='m'
             onClick={() => {
+              applyFieldsSelection();
               store.setDisplays(activeTab, selectedDisplays);
               setIsModalOpen(false);
             }}
```

Comment:
I've found a small bug :bug::
1. I open settings modal
2. I have all questions selected
3. I unselect some questions
4. I've decided I don't want to change it so I close modal (e.g. with ESC) 
5. I open settings modal again
6. I have some questions selected (it should be all questions selected, i.e. it should reflect what we have in store)

Reply:
I think a fix would be to apply `setSelectedFields(getInitialFields());` inside the `onRequestClose` function. This way after closing and reopening, the state would be fresh. Alternatively we could do it in the same function that is opening the modal (doing `setIsModalOpen(true)`). So either "cleanup afterwards" or "always start fresh" approach :D After doing either of these, please check if one of the places that resets its already is not needed (e.g. inside `resetFieldsSelection`)

Reply:
I think these are still needed in `resetFieldsSelection` as resetting is not the same as closing the modal--on reset, we need to tell the store to clear the hidden fields as well as reset them visually with `setSelectedFields(getInitialFields());`

Reply:
> I think this is a bug with the modal in general. Maybe we never caught it before but the same steps above can be taken for the regular toggle switches (the displays). And it's due to this: https://github.com/kobotoolbox/kpi/pull/4889/files/be15a3938a5956f36531d3e2325e0774cecfe635#diff-4e96bc318c90b92e52221bc45959e508183d0babeb08d081a876bcccdcc3d542L146-L149

I cleared this up by adding `setSelectedDisplays(store.getDisplays(activeTab))` into `onRequestClose` as well